### ISSUE TITLE
TypeScript support

### ISSUE THREAD

**Issue by asode** [kindoflew/svelte-parallax#26]
Hey there, just wanted to first thank you for the great library!

I tried the library out with my typescript project and it seems that the types are missing. Any chance of making this project typescript compatible?

My understanding of typescript is a bit limited but I was still able to get it to work and not throw any errors (at least for now) with the following declarations (which probably still need some work):

```node
// svelte-parallax.d.ts
declare module 'svelte-parallax' {
	import { SvelteComponentTyped } from 'svelte';

	interface ParallaxLayerProps {
		rate?: number = 0.5;
		offset?: number = 0;
		span?: number = 1;
		[key: string]: any; // for any other props passed to the underlying div container
	}
	export class ParallaxLayer extends SvelteComponentTyped<ParallaxLayerProps> {}

	interface ParallaxProps {
		sections?: number = 1;
		sectionHeight?: number = window.innerHeight;
		config?: {
			stiffness?: number = 0.017;
			damping?: number = 0.26;
		};
		threshold?: {
			top?: number = 1;
			bottom?: number = 1;
		};
		onProgress?: function = undefined;
		disabled?: boolean = false;
	}
	export class Parallax extends SvelteComponentTyped<ParallaxProps> {}
}
```

**Comment by kindoflew** [comment]
Hey there! Thanks for the kind words!

This is a great idea. I'm working on this now -- the only thing I need to figure out still is how I'll be typing the `scrollTo` method exported from `Parallax`.

Once that's sorted, I'll publish a new version with the declaration file included.

**Comment by kindoflew** [comment]
@asode this is now available in `0.5.2`

**Comment by asode** [comment]
That was fast! I tried it out and seems to work nicely, even the `scrollTo` method didn't give me any type errors. Thanks again!
